tests/kernel/mbox/mbox_api/src/test_mbox_api.c

```diff
@@ -564,6 +563,73 @@ void test_mbox_kdefine(void)
 	tmbox(&kmbox);
 }
 
+/**
+ *
+ * @brief Test mailbox enhance capabilities
+ *
+ * @details
+ * - Define and initilized a message queue and a mailbox
+ * - Verify the capability of message queue and mailbox
+ * - with same data.
+ *
+ */
+void test_enhance_capability(void)
+{
+	int MSG = 0xABCD;
+	struct k_msgq msgq;
+
+	ZTEST_BMEM char __aligned(4) buffer[8];
+	k_msgq_init(&msgq, buffer, 4, 2);
+	/* send buffer with message queue */
+	int ret = k_msgq_put(&msgq, &MSG, K_NO_WAIT);
+
+	zassert_false(ret, "message queue put successful");
+
+	/* send same buffer with mailbox */
+	sender_tid = k_current_get();
+	receiver_tid = k_thread_create(&tdata, tstack, STACK_SIZE,
+			tmbox_entry, &mbox, NULL, NULL,
+			K_PRIO_PREEMPT(0), 0, K_NO_WAIT);
+	struct k_mbox_msg send_msg;
+
+	send_msg.info = 4;
+	send_msg.size = sizeof(MSG);
+	send_msg.tx_data = &MSG;
+	send_msg.tx_block.data = NULL;
+	send_msg.tx_target_thread = receiver_tid;
+
+	k_mbox_put(&mbox, &send_msg, K_FOREVER);
+	zassert_equal(send_msg.size, sizeof(MSG), NULL);
```

Comment:
This... doesn't seem like a very useful test.  You create a mailbox, insert one message, and verify that the return code is correct.  You then abort the receiving thread (AFAICT before it even handles the message!) and return while checking a value in the send_msg struct that shouldn't have been modified anyway.  Almost all of that (and more) is already done by the existing tmbox() test.

What's the code path or requirement you're trying to add coverage for here?  It seems like there are parts missing.  At the very least you want to receive the message in the other thread and validate that it's the correct data, right?

Reply:
This case, I just want to show the mailbox is that provides enhanced message queue capabilities that go beyond the capabilities of a message queue object. Like this test, mailbox allow send messages of any size, message queue can't do that.
It is my purpose! Do you think it is ok?

Reply:
It seems to me that we need to make sure the messages sent via mbox or message queue has already received correctly.

Reply:
@andyross  @LeiW000 I fixed this issue. Do you have any other comments?